### PR TITLE
linuxPackages.ena: 2.8.3 -> 2.8.6 and fix build against Linux 6.3

### PR DESCRIPTION
**Pull request by edolstra**
###### Description of changes

This applies some patches that will be in the next version of ena, one fixing the removal of the adjfreq field in ptp_clock_info, and the others dealing with devlink changes.

https://hydra.nixos.org/build/221431596

Closes #224286 and #224287.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erdnaxe** [comment]
ZHF: https://github.com/NixOS/nixpkgs/issues/230712

This should get backported to release-23.05.

File: pkgs/os-specific/linux/ena/default.nix [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
-  version = "2.8.3";
+  version = "2.8.6";
   name = "ena-${version}-${kernel.version}";
 
   src = fetchFromGitHub {
     owner = "amzn";
     repo = "amzn-drivers";
     rev = "ena_linux_${version}";
-    sha256 = "sha256-Z5BCqqC4FNKHqpocJ6CsHwhh5sszj5QfxiN6+Dkaqek=";
+    sha256 = "sha256-clRu+ecK/Je0kvlAAm6qCJqMyvZv0C88YIGDImhRhKA=";
```
Comment (erdnaxe) [inline pkgs/os-specific/linux/ena/default.nix:11]:
```suggestion
    hash = "sha256-clRu+ecK/Je0kvlAAm6qCJqMyvZv0C88YIGDImhRhKA=";
```

File: pkgs/os-specific/linux/ena/default.nix [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
+  patches =
+    [ # https://github.com/amzn/amzn-drivers/issues/269#issuecomment-1552483792
+      (fetchpatch {
+        url = "https://github.com/amzn/amzn-drivers/files/11504862/phc_kernel_6_2_fix.patch";
+        hash = "sha256-/EBkISwXMd7t4WZjsG9KVP6vncFwcZq1QBsxQLXyWsY=";
+      })
+      # https://github.com/amzn/amzn-drivers/issues/270#issuecomment-1561924754
+      (fetchpatch {
+        url = "https://github.com/amzn/amzn-drivers/files/11559312/devlink_6_2_fix.patch";
+        hash = "sha256-J7CnW3d2w3qFTH9jPBsOszOuBB2AWLyvKruLhO+YiWA=";
```
Comment (erdnaxe) [inline pkgs/os-specific/linux/ena/default.nix:23]:
```suggestion
        hash = "sha256-Nc71u91G0dL+ld6ovqjHaE6X2TxduWeQYr5K0KdoA3Q=";
```

File: pkgs/os-specific/linux/ena/default.nix [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
+      })
+      (fetchpatch {
+        url = "https://github.com/amzn/amzn-drivers/files/11559314/devlink_6_3_fix.patch";
+        hash = "sha256-1ArmJyotZQm+9fLdiQcT+3oCmjzUfJn2saWVU0vA/pg=";
```
Comment (erdnaxe) [inline pkgs/os-specific/linux/ena/default.nix:27]:
```suggestion
        hash = "sha256-aEQTbwHC1DcDrtj188eoGzi3GU9MXnwIxuJW4L7qb/I=";
```

**Review by erdnaxe — CHANGES_REQUESTED** [review]
LGTM, thanks!
Please fix the hashs as ofborg failed to build this PR.

(please tag ZHF when opening such pull request)

**Comment by edolstra** [comment]
Weird, the hashes work for me...

**Review by sielicki — APPROVED** [review]
LGTM.

**Pull request merged**